misc-bench/cpp-relu-batch/relu_batch.cpp

```diff
@@ -0,0 +1,339 @@
+  int numRows = inputMatrix[0].size();
+  int numCols = inputMatrix[0][0].size();
+  
+  // Initialize the output matrix with zeros
+  std::vector<std::vector<std::vector<int>>> outputMatrix(numDepth, std::vector<std::vector<int>>(numRows, std::vector<int>(numCols, 0)));
+       
+  int mismatch_counter = 0;
+  // Perform RELU with the specified kernel size and stride
+  for (int d = 0; d < numDepth; ++d) {
+    for (int i = 0; i < numRows; ++i) {
+      for (int j = 0; j < numCols; ++j) {
+        outputMatrix[d][i][j] = std::max(0, inputMatrix[d][i][j]);
+        if (outputMatrix[d][i][j] != PIMResult[d][i][j]) {
+          std::cout << "Mismatch between PIM and CPU results at depth: " << d << ", row: " << i << ", column: " << j << std::endl;
+          mismatch_counter += 1;
+        }        
+      }
+    }
+  }
+   
+  if (mismatch_counter == 0) {
+    std::cout << "Success: PIM results match with CPU results" << std::endl << std::endl; 
+  } else {
+    std::cout << "Failure: PIM results do not match with CPU results" << std::endl << std::endl;
+  }
+
+  if (moreDebugPrints == true) {
+    std::cout << "Input matrix:" << std::endl;
+    printMatrix(inputMatrix);
+    std::cout << "Output matrix from CPU:" << std::endl;
+    printMatrix(outputMatrix);
+    std::cout << "Output matrix from PIM:" << std::endl;
+    printMatrix(PIMResult);    
+  }
+  
+  return outputMatrix;
+}
+
+int main(int argc, char *argv[]) {
+
+  // Parse input parameters from command line
+  Params params = getInputParams(argc, argv);
+
+  // Initialize input matrix based on parsed dimensions
+  std::vector<std::vector<std::vector<int>>> inputMatrix(params.dim, std::vector<std::vector<int>>(params.row, std::vector<int>(params.column * params.batchSize)));
+
+  // Check if an image matrix file is provided
+  if (params.imageMatrixFile == nullptr) {
+    // Generate or retrieve matrix data if no file is provided
+    for (auto &mat : inputMatrix) {
+      getMatrix(params.row, (params.column * params.batchSize), 0, mat);
+    }
+  } else {
+    std::cout << "Reading from input file is not implemented yet." << std::endl;
+    return 0;    
+  }
+
+  // Create a device based on provided DRAM configuration file
+  if (!createDevice(params.dramConfigFile))
+    return 1;
+
+
+  // Calculate matrix dimensions
+  int inputDepth = inputMatrix.size();
+  int inputHeight = inputMatrix[0].size();
+  int inputWidth = inputMatrix[0][0].size();
+
+  // Define parameters for processing
+  PimDeviceProperties deviceProp;
+  PimStatus status = pimGetDeviceProperties(&deviceProp);
+  if (status != PIM_OK) {
+    std::cout << "Abort: pimGetDeviceProperties failed" << std::endl;
+    return 1;
+  }
+  // Get the device parameters
+  uint64_t numCols = uint64_t (deviceProp.numColPerSubarray);
+  uint64_t numRows = uint64_t (deviceProp.numRowPerSubarray);
+  uint64_t numOfBits = uint64_t(deviceProp.numRanks) * uint64_t(deviceProp.numBankPerRank) * uint64_t (deviceProp.numSubarrayPerBank);
```

Comment:
Shouldn't you multiply this wit rows and columns too?

Reply:
So, the multiplication of numofBits, numCols and numRows is done in the next step for numOfMatPerRow. Will update the variable names.